float64[] coeff
int8 poly_order


